Landscape and portrait

nunoh opened the pull request.

nunoh committed:
Implements both portrait and landscape mode (#89)

Had to restructure the code quite a bit for this one. The difficult part
was to be able to maintain the state (position of the particles
basically) while also drawing everything correctly. It is a simple as
having a different update cycle when orientation happens. The way to
manage this easily in code was to encapsulate as much behaviour as
possible inside the classes (Module, ModuleConsole, Button and Fader)
and outside the main ofApp class with little repetition. For that I used
a .setDimensions() method "pattern" that is both called by the
constructor of the class when starting the app as well as when
orientation changes. I do that for all the four aforementioned classes
and that is pretty much it.

Apart from that, so minor bug fixes included:

- the modules were starting with wrong values of y, namely the first one
  with y=0, the second one with y=1, and the following two ones with y=2
  and y=3, which generated some minor yet noticeably visible artifacts.

- clearer convention for defines and helper defines too;

- better type safe signatures methods: was using float when should
  clearly be int;

- removed some obsolete code: icons stuff, different drawing routines,
  etc.

nunoh committed:
Closes #89

Oscar-Rodrigues commented:
So, just for keeping everything organized, this is how we should switch between portrait and landscape modes on the iPad/Android tablets:
1. Open in Landscape mode. All modules functional;
2. Register the module we last interacted with as the "active" module in landscape. When no interaction has taken place, if defaults to the first module;
3. When switching to Portrait, focus on the "active" module and freeze/pause all others. Only the active module plays;
4. When switching back to landscape, all modules are unfreezed/unpaused and they begin playing again.

nunoh committed:
Multiple fixes

- remove unnecessary files in bin/data folder in order to save on final
binary. That includes, not including TuioRouter folder for an iOS
release for instance. We are currently copying asserts from project,
since still haven't found a good symlink solution;

- using astyle so that code looks nice when previewing in Github;

- refactored to #if defined everywhere;

- ofLog doesn't need `endl` at the end;

- changed ofLog calls so that don't change level at every one;

- different `drawModules()` according to platform;

- forward declaration of class Touch fix;

- detect if iPad or iPhone at runtime, since the variable we were using
of `TARGET_OF_IPHONE` is in version 0.9.3 being defined for both when in
iphone and ipad.

nunoh committed:
It's `TARGET_OSX` and not `TARGET_OF_OSX`

nunoh committed:
Makes iPad version start in landscape

nunoh committed:
Finally closing PR

- Added logic for detecting last active module
- Fix for down arrow working in landscape
- Fix bug in setting particles position on orientation change

nunoh committed:
Merge branch 'master' into landscape-and-portrait

Make sure to integrate newer changes in master into feature branch, so
that PR merges ok.

nunoh committed:
Needed to update project references for Cocos addon

nunoh committed:
Fixes #94 (fader size in portrait)

nunoh committed:
Fix #105 (multitouch problem)

nunoh committed:
Fixes for iPad

It is quite a problem not being able to detect if running for iPad or
iPhone in the preprocessing stage.. As of right now, because the modules
are a static variable, even for iPhone we need to initialize with 4
modules, en though we just need one. Should try and get a fix for that.
